From 58fde75dccd5e06a22078143e17b03cca8be14d2 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 17 Jan 2013 13:52:53 +0800
Subject: [PATCH 06/32] ENGR00238394 Ethernet:delete link up update in
 StartMonitoring

Indicate link up state should only be decided by interface config after
startmonitoring.If no ethernet cable plug in, in startmonitoring,it will
set link up to true. When we plug in ethernet first time, it will fail to
process the link state change for it not satisfy "mLinkUp != up" condition.
So link up state should only based on the interface config which come from
the link state change observer.

Change-Id: Iadcdc247751f9bee9b1281b1678b8520830d321d
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 core/java/android/net/EthernetDataTracker.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/core/java/android/net/EthernetDataTracker.java b/core/java/android/net/EthernetDataTracker.java
index 5229328..338db1b 100644
--- a/core/java/android/net/EthernetDataTracker.java
+++ b/core/java/android/net/EthernetDataTracker.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2010 The Android Open Source Project
- * Copyright (C) 2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -231,7 +231,6 @@ public class EthernetDataTracker extends BaseNetworkStateTracker {
                     mIface = iface;
                     mNMService.setInterfaceUp(iface);
                     InterfaceConfiguration config = mNMService.getInterfaceConfig(iface);
-                    mLinkUp = config.hasFlag("up");
                     if (config != null && mHwAddr == null) {
                         mHwAddr = config.getHardwareAddress();
                         if (mHwAddr != null) {
-- 
1.8.0

